lib/timers/events.rb

```diff
@@ -60,18 +60,12 @@ def schedule(time, callback)
       # Maintain sorted order, O(logN) insertion time.
       @sequence.insert(index, handle)
       
-      return handle
+      handle
     end
     
     # Returns the first non-cancelled handle.
     def first
-      while handle = @sequence.last
-        if handle.cancelled?
-          @sequence.pop
-        else
-          return handle
-        end
-      end
+      @sequence.reverse.find { |handle| !handle.cancelled? }
```

Comment:
This seems like it both changes the behavior (it's no longer popping the cancelled handles) and the `#reverse` operation is O(n) for n handles.

Did you check how this affected the benchmarks?